### ISSUE TITLE
Comparison between Nanolog

### ISSUE THREAD

**Issue by oPiZiL** [Morgan-Stanley/binlog#100]
https://github.com/PlatformLab/NanoLog

**Comment by erenon** [comment]
Are you looking for a comparison? From which aspect? Runtime or compilation performance, features, loggable types, supported platforms, documentation, ...?

**Comment by oPiZiL** [comment]
Yes, a comparison.  mainly about performance, especially with multi-thread support

**Comment by lawrence-koh** [comment]
@oPiZiL  It seems like the main benefit of using binlog over NanoLog is that binlog does not depend on a background thread and provides the user with full control over when to `consume` the log lines produced, allowing the library to be used in strictly single-threaded applications.

In fact this is also raised as an issue for NanoLog (https://github.com/PlatformLab/NanoLog/issues/43).

@erenon just to check if my understanding is correct?

**Comment by erenon** [comment]
In that sense, binlog works as you described it: the application (user) is in control of the consuming.
There are probably other differences, but I don't know nanolog enough to list them. I can confirm binlog features though.

**Comment by lawrence-koh** [comment]
@erenon sorry I wasn't clear, was just looking for your help to confirm how binlog works. Thanks!
